File: spine-libgdx/src/com/esotericsoftware/spine/SlotData.java
/*******************************************************************************
 * Copyright (c) 2013, Esoteric Software
 * All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 * 
 * 1. Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
 * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
 * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ******************************************************************************/

package com.esotericsoftware.spine;

import com.badlogic.gdx.graphics.Color;

public class SlotData {
	final String name;
	final BoneData boneData;
	final Color color = new Color(1, 1, 1, 1);
	String attachmentName;
	boolean additiveBlending;

	SlotData () {
		name = null;
		boneData = null;
	}

	public SlotData (String name, BoneData boneData) {
		if (name == null) throw new IllegalArgumentException("name cannot be null.");
		if (boneData == null) throw new IllegalArgumentException("boneData cannot be null.");
		this.name = name;
		this.boneData = boneData;
	}

	public String getName () {
		return name;
	}

	public BoneData getBoneData () {
		return boneData;
	}

	public Color getColor () {
		return color;
	}

	/** @param attachmentName May be null. */
	public void setAttachmentName (String attachmentName) {
		this.attachmentName = attachmentName;
	}

	/** @return May be null. */
	public String getAttachmentName () {
		return attachmentName;
	}

	public boolean getAdditiveBlending () {
		return additiveBlending;
	}

	public void setAdditiveBlending (boolean additiveBlending) {
		this.additiveBlending = additiveBlending;
	}

	public String toString () {
		return name;
	}
}

File: spine-cpp/spine-cpp/src/spine/SlotData.cpp
/******************************************************************************
* Spine Runtimes Software License v2.5
*
* Copyright (c) 2013-2016, Esoteric Software
* All rights reserved.
*
* You are granted a perpetual, non-exclusive, non-sublicensable, and
* non-transferable license to use, install, execute, and perform the Spine
* Runtimes software and derivative works solely for personal or internal
* use. Without the written permission of Esoteric Software (see Section 2 of
* the Spine Software License Agreement), you may not (a) modify, translate,
* adapt, or develop new applications using the Spine Runtimes or otherwise
* create derivative works or improvements of the Spine Runtimes or (b) remove,
* delete, alter, or obscure any trademarks or any copyright, trademark, patent,
* or other intellectual property or proprietary rights notices on or in the
* Software, including any copy thereof. Redistributions in binary or source
* form must include this license and terms.
*
* THIS SOFTWARE IS PROVIDED BY ESOTERIC SOFTWARE "AS IS" AND ANY EXPRESS OR
* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
* MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
* EVENT SHALL ESOTERIC SOFTWARE BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
* PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES, BUSINESS INTERRUPTION, OR LOSS OF
* USE, DATA, OR PROFITS) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
* IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
* POSSIBILITY OF SUCH DAMAGE.
*****************************************************************************/

#include <spine/SlotData.h>

#include <assert.h>

namespace Spine
{
    SlotData::SlotData(int index, std::string name, BoneData& boneData) :
    _index(index),
    _name(name),
    _boneData(boneData),
    _r(1),
    _g(1),
    _b(1),
    _a(1),
    _r2(0),
    _g2(0),
    _b2(0),
    _a2(1),
    _hasSecondColor(false),
    _attachmentName(),
    _blendMode(BlendMode_Normal) {
        assert(_index >= 0);
        assert(_name.length() > 0);
    }
    
    const int SlotData::getIndex() {
        return _index;
    }
    
    const std::string& SlotData::getName() {
        return _name;
    }
    
    BoneData& SlotData::getBoneData() {
        return _boneData;
    }
    
    float SlotData::getR() {
        return _r;
    }
    
    void SlotData::setR(float inValue) {
        _r = inValue;
    }
    
    float SlotData::getG() {
        return _g;
    }
    
    void SlotData::setG(float inValue) {
        _g = inValue;
    }
    
    float SlotData::getB() {
        return _b;
    }
    
    void SlotData::setB(float inValue) {
        _b = inValue;
    }
    
    float SlotData::getA() {
        return _a;
    }
    
    void SlotData::setA(float inValue) {
        _a = inValue;
    }
    
    float SlotData::getR2() {
        return _r2;
    }
    
    void SlotData::setR2(float inValue) {
        _r2 = inValue;
    }
    
    float SlotData::getG2() {
        return _g2;
    }
    
    void SlotData::setG2(float inValue) {
        _g2 = inValue;
    }
    
    float SlotData::getB2() {
        return _b2;
    }
    
    void SlotData::setB2(float inValue) {
        _b2 = inValue;
    }
    
    bool SlotData::hasSecondColor() {
        return _hasSecondColor;
    }
    
    void SlotData::setHasSecondColor(bool inValue) {
        _hasSecondColor = inValue;
    }
    
    std::string SlotData::getAttachmentName() {
        return _attachmentName;
    }
    
    void SlotData::setAttachmentName(std::string inValue) {
        _attachmentName = inValue;
    }
    
    BlendMode SlotData::getBlendMode() {
        return _blendMode;
    }
    
    void SlotData::setBlendMode(BlendMode inValue) {
        _blendMode = inValue;
    }
}
